### DISCUSSION TITLE
Battery life

### DISCUSSION THREAD

**Discussion by d3multimedia** [RonMcKay/capacitive-soil-moisture-sensor#4] [Q&A] [answered]
With the current iteration of the board, what is the runtime on 2AAA batteries?

**Comment by RonMcKay** [comment, marked as answer]
I never calculated an estimated battery lifetime but that would indeed be very interesting.
In the 2-3 months that I had the first prototype running I did not observe any significant drop in battery voltage. That obviously depends strongly on the sampling interval (I set it to 30 minutes).

When I have some spare time in the next few days I will take some power consumption measurements and report it back here. Then we will get an approximate battery lifetime.

**Reply by RonMcKay** [reply]
So I did some measurements and these are the current consumption values:

measuring: ~3.8 mA (takes ~250 ms)
transmitting: ~ 17.6 mA (takes ~50ms???, just a guess...)
sleeping: ~ 8 uA (30 min)

Now the battery runtime obviously is influenced by a lot of factors, e.g. like ambient temperature (cold weather outside reduces battery capacity), measuring interval (I calculate with the default 30min sleep interval) and quality of the battery.

It was hard to find a decent chart but I used this one for my calculations:

[Discharge capacity Duracell Plus Power AAA](https://lygte-info.dk/pic/BatteryLowCurrentDischarge/AA-AAA/Discharge%20capacity%20Duracell%20Plus%20Power%20AAA.png)

Here I am assuming that we have 1000mAh before we reach 1.2V on the battery. This is the lower end since the ATmega328P at 8 MHz only works reliably down to 2.4V.

Based on that the optimal battery life of the sensor is around 10 years. In reality you will not get there as the self discharge rate of the AAA are probably higher than the sleeping current of the sensor but I guess you can expect a few years of battery life with this sensor.
